.github/workflows/run_periodic_tests.yml

```diff
@@ -102,7 +102,7 @@ jobs:
 
       - name: Upload coverage report
         if: matrix.os == 'ubuntu-latest' && matrix.python-version == 3.11
-        uses: codecov/codecov-action@v2.1.0
+        uses: codecov/codecov-action@v3.1.4
```

Comment:
These versions go stale very often in PyBaMM. I have opened #3276 for this.

Reply:
Awesome!